VRDR.Messaging/CodingResponseMessage.cs

```diff
@@ -140,14 +140,267 @@ public enum RaceCode
                 var list = new List<Tuple<string, Base>>();
                 foreach (KeyValuePair<RaceCode, string> item in value)
                 {
-                    var part = Tuple.Create(item.Key.ToString(), 
-                        (Base)(new Coding("https://www.cdc.gov/nchs/data/dvs/RaceCodeList.pdf", item.Value)));
+                    string codeList = "https://www.cdc.gov/nchs/data/dvs/RaceCodeList.pdf";
+                    // Special case for RACEBRG, it uses a different code document
+                    if(item.Key == RaceCode.RACEBRG) {
+                        codeList = "https://www.cdc.gov/nchs/data/dvs/Multiple_race_documentation_5-10-04.pdf";
+                    }
+                    var part = Tuple.Create(item.Key.ToString(),
+                        (Base)(new Coding(codeList, item.Value)));
                     list.Add(part);
                 }
                 Record.Add("race", list);
             }
         }
 
+        /// <summary>Coder Status (Numeric, 0-9)</summary>
+        public uint? CoderStatus
+        {
+            get
+            {
+                return (uint?)Record.GetSingleValue<UnsignedInt>("cs")?.Value;
+            }
+            set
+            {
+                Record.Remove("cs");
+                if (value != null)
+                {
+                    if (value < 0 || value > 9) {
+                        throw new ArgumentException("Valid values for CoderStatus are 0-9 or null");
+                    }
+                    Record.Add("cs", new UnsignedInt((int)value));
+                }
+            }
+        }
+
+        /// <summary>Shipment Number (Alpha Numeric)</summary>
+        public string ShipmentNumber
+        {
+            get
+            {
+                return Record.GetSingleValue<FhirString>("ship")?.Value;
+            }
+            set
+            {
+                Record.Remove("ship");
+                if (value != null)
+                {
+                    Record.Add("ship", new FhirString(value));
+                }
+            }
+        }
+
+        /// <summary>NCHS Receipt Date Month (Numeric, 01-12 or null)</summary>
```

Comment:
May want to check with users to see if a uint is going to require them to do string padding (e.g. 1 -> "01") for their needs. May want to switch to a string or offer a convenience method if so. Same for NCHSReceiptDay and (maybe for consistency) year.